File: dynamic-programming-level-1/day-2/s2.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        p = [cost[0], cost[1]]
        for i in range(2, len(cost)):
            p.append(cost[i] + min(p[i-1], p[i-2]))
        return min(p[-2], p[-1])

File: dynamic-programming-level-1/day-2/s2.c
int minCostClimbingStairs(int *cost, int costSize)
{
    int p[costSize];
    p[0] = cost[0];
    p[1] = cost[1];
    for (int i = 2; i < costSize; i++)
    {
        if (p[i - 2] < p[i - 1])
            p[i] = cost[i] + p[i - 2];
        else
            p[i] = cost[i] + p[i - 1];
    }
    return (p[costSize - 1] < p[costSize - 2]) ? p[costSize - 1] : p[costSize - 2];
}